### PR TITLE
Fixes mushroom bowls

### LINKED ISSUE #32471: Mushroom bowls can't be used as regular bowls in tablecrafting + reagents won't spill when thrown.
Can't make the most important meal of the day, wish soup.

Also can't throw them to spill reagents on the floor or yourself.

### PR DESCRIPTION
**Pull request by fludd12**
Made the mushroom bowl a subtype of the normal bowl, and deleted a
duplicate definition.

Also, adds custom soup and salad support for the mushroom bowls.

Fixes #32471

### CODE REVIEW

**Comment by praisenarsie** [comment]
### ITS THE REAL FLUDD12

**Comment by fludd12** [comment]
...Was there a fake fludd12...?

**Comment by fludd12** [comment]
@optimumtact Does it look good?

File: code/modules/mining/lavaland/ash_flora.dm [hunk 1]
```diff
@@ -191,20 +191,55 @@
...
+		else
+			if(reagents.has_reagent("water", 10)) //are we starting a soup or a salad?
+				var/obj/item/reagent_containers/food/snacks/customizable/A = new/obj/item/reagent_containers/food/snacks/customizable/soup/ashsoup(get_turf(src))
+				A.initialize_custom_food(src, S, user)
+			else
+				var/obj/item/reagent_containers/food/snacks/customizable/A = new/obj/item/reagent_containers/food/snacks/customizable/salad/ashsalad(get_turf(src))
+				A.initialize_custom_food(src, S, user)
+	else
+		. = ..()
+	return
```
Comment (ShizCalev) [inline code/modules/mining/lavaland/ash_flora.dm]: don't need this return

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by fludd12** [comment]
Ah, thanks for the fix, I didn't have power yesterday.

**Pull request merged**